mat_dp_pipeline/sdf_to_input.py

```diff
@@ -236,5 +236,6 @@ def validate_sdf(root: Path):
     Raises: when parsing of the SDF fails at some point.
     """
     # Pass `root` through `sdf_to_combined_input`, ignoring the result of it
-    for _ in sdf_to_combined_input(sdf.load(root)):
-        pass
+    # TODO: This means pipeline calls sdf_to_combined_input twice - not ideal,
+    # could either solve with caching or alternatively is this check needed here?
+    list(sdf_to_combined_input(sdf.load(root)))
```

Comment:
Yeah, might be not needed... This was before DataSource would give SDF instead of just writing to a dir. Now sdf.load does actually most of the validation I think...

Reply:
Ah I see - what steps does sdf.load not do but validate does?

Reply:
Maybe I'm talking nonsense. validate does actually a lot more, but it could do more if we pushed it further through the "pipeline". I guess it should be removed... validation will happen as its processed. Unless we can come up with some basic validation mechanism which doesn't execute all of the pipeline.